wandb/integration/yolov8/yolov8.py

```diff
@@ -184,6 +196,41 @@ def teardown(self, _trainer: BaseTrainer) -> None:
         self.run.finish()
         self.run = None
 
+    def on_val_start(
+        self,
+        validator: BaseValidator,
+    ) -> None:
+        """On validation start we create a wandb table to store the sample predictions
+        and patch the validator's `plot_predictions` method to plot the predictions to the wandb table .
+        """
+        class_names = sorted(validator.names.values())
+        if isinstance(validator, DetectionValidator):
+            self.validation_table = wandb.Table(
+                columns=["Batch", "Ground Truth", "Prediction", *class_names]
+            )
+            self.og_plot_predictions = validator.plot_predictions
+            validator.plot_predictions = partial(
+                plot_detection_predictions, self, validator
+            )
+
+        if isinstance(validator, SegmentationValidator):
+            self.validation_table = wandb.Table(
+                columns=["Batch", "Ground Truth", "Prediction", *class_names]
+            )
+            self.og_plot_predictions = validator.plot_predictions
```

Comment:
can we rename it to something more clear `og_plot_predictions`?

Reply:
changed this to `original_plot_predictions` as recommended